File: 48. Rotate Image/48. Rotate Image.py
class Solution(object):
    def rotate(self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: None Do not return anything, modify matrix in-place instead.
        """
        new=[]
        for i in range(len(matrix)):
            temp=[]
            for j in range(1,len(matrix)+1):
                temp.append(matrix[-j][i])
            new.append(temp)
        matrix[:]=new

File: 48. Rotate Image.java
class Solution {
    public void reverse(int[] matrix){
        int n=matrix.length-1;
        int m=0;
        while(m<n){
            int temp=matrix[m];
            matrix[m]=matrix[n];
            matrix[n]=temp;
            m++;
            n--;
        }
    }

    public void rotate(int[][] matrix) {
        for(int i=0;i<matrix.length;i++)
        {
            for(int j=i;j<matrix.length;j++)
            {
                int temp=matrix[i][j];
                matrix[i][j]=matrix[j][i];
                matrix[j][i]=temp;
            }
        }
        for(int i=0;i<matrix.length;i++)
        {
            reverse(matrix[i]);

        }
    }
}
